ui/pages/swaps/prepare-swap-page/review-quote.js

```diff
@@ -880,7 +912,7 @@ export default function ReviewQuote({ setReceiveToAmount }) {
   ]);
 
   useEffect(() => {
-    if (isSmartTransaction && !insufficientTokens) {
+    if (isSmartTransaction && !insufficientTokens && !isGasIncludedTrade) {
```

Comment:
A comment like this might help readability here:
// if it's a smart transaction, has sufficient tokens, and gas is not included in the trade,
// set up gas fee polling.